content/en/platform/corda/4.11/enterprise/finality-flow-recovery.md

```diff
@@ -0,0 +1,468 @@
+* `peers` field contains the other participants' distribution list.
+{{< /note >}}
+
+You can also perform the same check by specifying a transaction ID.  This is commonly the case when recovering from a receiver perspective (as an initiator flow ID is meaningless outside its own node).
+
+```bash
+flowStatus queryFinalityByTxnId 19BE64484D3CBF532A8FB2ACA1AEACA38B1FBA3C38B0518B7F5316AC9E79432F
+```
+
+To check the status of a flow transaction as a `ReceiveFinalityFlow` receiver:
+
+```bash
+flowStatus queryFinalityByTxnId 7E7EE31CA6371D73CDDB1A7992E239CE222606EA845F1ABC87995898017904A4
+
+FlowTransactionInfo(stateMachineRunId=[8cc13ed6-ea14-43e6-a7b0-85a61fb3bbb1], txId=7E7EE31CA6371D73CDDB1A7992E239CE222606EA845F1ABC87995898017904A4, status=IN_FLIGHT, timestamp=2023-03-29T12:40:23.628Z, initiator=O=Alice Corp, L=Madrid, C=ES, peers=null)
+---
+- stateMachineRunId:
+    uuid: "8cc13ed6-ea14-43e6-a7b0-85a61fb3bbb1"
+  txId: "7E7EE31CA6371D73CDDB1A7992E239CE222606EA845F1ABC87995898017904A4"
+  status: "IN_FLIGHT"
+  initiator:
+    x500Principal:
+      name: "O=Alice Corp,L=Madrid,C=ES"
+  peers: null
+```
+
+{{< note >}}
+* `initiator` refers to the party that initiated the `FinalityFlow`.
+* `peers` field is **null**: this information is private to the initiator only.
+{{< /note >}}
+
+To recover a failed finality flow using the flow ID:
+
+```bash
+flow recoverFinality 821884be-8e9f-486d-8228-70d97e215218
+Recovered finality flow [821884be-8e9f-486d-8228-70d97e215218]
+```
+
+Should recovery fail, you will see the following response:
+
+```bash
+Failed to recover finality flow 821884be-8e9f-486d-8228-70d97e215218
+```
+
+Further information explaining why a flow failed to recover can be found in the node logs.
+A prime example is attempting to recover a flow that has already completed successfully.
+The node logs will show the following message:
+
+```bash
+Recovery not possible for transaction with status VERIFIED
+```
+
+To recover a failed finality flow by using the transaction ID:
+
+```bash
+flow recoverFinalityByTxnId 7E7EE31CA6371D73CDDB1A7992E239CE222606EA845F1ABC87995898017904A4
+Recovered finality flow [821884be-8e9f-486d-8228-70d97e215218]
+```
+Should recovery fail, you will see the following response:
+
+```bash
+Failed to recover finality flow 7E7EE31CA6371D73CDDB1A7992E239CE222606EA845F1ABC87995898017904A4
+```
+
+After successfully recovering a finality flow, the flow transaction status should move to `VERIFIED`.
+
+To recover all failed finality flows in one operation:
+
+```bash
+flow recoverAllFinality
+Recovered finality flow(s)
+Results: [[4cbfa031-90de-4564-a375-30141a18bbba]=true]
+```
+
+{{< note >}}
+This operation returns a list of flow identifier/Boolean (success/failure) pairs.
+In all other cases, the shell prints a message stating if the operation succeeded or not.
+{{< /note >}}
+
+To recover all failed finality flows, including those in a PAUSED or HOSPITALIZED checkpoint state:

```

Comment:
Should PAUSED and HOSPITALIZED be in code? Think they were previously. Maybe check throughout.

Reply:
Done